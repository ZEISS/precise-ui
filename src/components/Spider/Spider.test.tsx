import * as React from 'react';
import * as enzyme from 'enzyme';
import { Spider } from './';

describe('<Spider />', () => {
  it('should render <Spider>', () => {
    const wrapper = enzyme.shallow(<Spider />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show two simple nodes in the <Spider>', () => {
    const nodes = [
      {
        title: 'Some title',
        head: 'Some head',
      },
      {
        title: 'Other title',
        head: 'Other head',
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show two connected nodes in the <Spider>', () => {
    const nodes = [
      {
        title: 'Some title',
        head: 'Some head',
        connected: 'other',
      },
      {
        id: 'other',
        title: 'Other title',
        head: 'Other head',
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show two child nodes in the <Spider>', () => {
    const nodes = [
      {
        title: <b>Foo</b>,
        head: 'Some head',
      },
      {
        title: <b>Bar</b>,
        head: (
          <h2>
            My <u>head</u>
          </h2>
        ),
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show three simple nodes using custom spacing in the <Spider>', () => {
    const nodes = [
      {
        title: 'Some title',
        head: 'Some head',
      },
      {
        title: 'Other title',
        head: 'Other head',
      },
      {
        title: 'Last title',
        head: 'Last head',
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} nodeSpacing={20} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show three simple nodes using node width in the <Spider>', () => {
    const nodes = [
      {
        title: 'Some title',
        head: 'Some head',
      },
      {
        title: 'Other title',
        head: 'Other head',
      },
      {
        title: 'Last title',
        head: 'Last head',
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} nodeWidth={200} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show five connected nodes in the <Spider>', () => {
    const nodes = [
      {
        id: '1',
        title: 'First',
        head: 'The head',
        connected: '5',
      },
      {
        id: '2',
        title: 'Second',
        head: 'The head',
        connected: '4',
      },
      {
        id: '3',
        title: 'Third',
        head: 'The head',
        connected: '4',
      },
      {
        id: '4',
        title: 'Fourth',
        head: 'The head',
      },
      {
        id: '5',
        title: 'Fifth',
        head: 'The head',
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show two connected nodes with body in the <Spider>', () => {
    const nodes = [
      {
        title: 'Some title',
        head: 'Some head',
        body: <div>Foo</div>,
        connected: 'other',
      },
      {
        id: 'other',
        title: 'Other title',
        body: 'bar',
        head: 'Other head',
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show a node with body elements in the <Spider>', () => {
    const nodes = [
      {
        title: 'Some title',
        head: 'Some head',
        body: [<div key="foo">Foo</div>, 'Hello', <span key="final">Final</span>],
      },
    ];
    const wrapper = enzyme.shallow(<Spider nodes={nodes} />);
    expect(wrapper).toMatchSnapshot();
  });
});
