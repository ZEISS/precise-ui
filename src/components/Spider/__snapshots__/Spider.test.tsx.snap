// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spider /> should render <Spider> 1`] = `
<styled.div>
  <styled.div
    width={-30}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={0}
      rows={1}
      spacing={30}
    />
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show a node with body elements in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={150}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={1}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="0"
        width={150}
      >
        <styled.div>
          Some title
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
        <styled.div>
          <div
            key="foo"
          >
            Foo
          </div>
          Hello
          <span
            key="final"
          >
            Final
          </span>
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show five connected nodes in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={870}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={5}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="1"
        width={150}
      >
        <styled.div>
          First
        </styled.div>
        <styled.div>
          The head
        </styled.div>
      </styled.div>
      <styled.div
        key="2"
        width={150}
      >
        <styled.div>
          Second
        </styled.div>
        <styled.div>
          The head
        </styled.div>
      </styled.div>
      <styled.div
        key="3"
        width={150}
      >
        <styled.div>
          Third
        </styled.div>
        <styled.div>
          The head
        </styled.div>
      </styled.div>
      <styled.div
        key="4"
        width={150}
      >
        <styled.div>
          Fourth
        </styled.div>
        <styled.div>
          The head
        </styled.div>
      </styled.div>
      <styled.div
        key="5"
        width={150}
      >
        <styled.div>
          Fifth
        </styled.div>
        <styled.div>
          The head
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show three simple nodes using custom spacing in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={490}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={3}
      rows={1}
      spacing={20}
    >
      <styled.div
        key="0"
        width={150}
      >
        <styled.div>
          Some title
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
      </styled.div>
      <styled.div
        key="1"
        width={150}
      >
        <styled.div>
          Other title
        </styled.div>
        <styled.div>
          Other head
        </styled.div>
      </styled.div>
      <styled.div
        key="2"
        width={150}
      >
        <styled.div>
          Last title
        </styled.div>
        <styled.div>
          Last head
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show three simple nodes using node width in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={660}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={3}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="0"
        width={200}
      >
        <styled.div>
          Some title
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
      </styled.div>
      <styled.div
        key="1"
        width={200}
      >
        <styled.div>
          Other title
        </styled.div>
        <styled.div>
          Other head
        </styled.div>
      </styled.div>
      <styled.div
        key="2"
        width={200}
      >
        <styled.div>
          Last title
        </styled.div>
        <styled.div>
          Last head
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show two child nodes in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={330}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={2}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="0"
        width={150}
      >
        <styled.div>
          <b>
            Foo
          </b>
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
      </styled.div>
      <styled.div
        key="1"
        width={150}
      >
        <styled.div>
          <b>
            Bar
          </b>
        </styled.div>
        <styled.div>
          <h2>
            My 
            <u>
              head
            </u>
          </h2>
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show two connected nodes in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={330}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={2}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="0"
        width={150}
      >
        <styled.div>
          Some title
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
      </styled.div>
      <styled.div
        key="other"
        width={150}
      >
        <styled.div>
          Other title
        </styled.div>
        <styled.div>
          Other head
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show two connected nodes with body in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={330}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={2}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="0"
        width={150}
      >
        <styled.div>
          Some title
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
        <styled.div>
          <div>
            Foo
          </div>
        </styled.div>
      </styled.div>
      <styled.div
        key="other"
        width={150}
      >
        <styled.div>
          Other title
        </styled.div>
        <styled.div>
          Other head
        </styled.div>
        <styled.div>
          bar
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;

exports[`<Spider /> should show two simple nodes in the <Spider> 1`] = `
<styled.div>
  <styled.div
    width={330}
  >
    <styled.canvas
      innerRef={[Function]}
    />
    <Styled(Grid)
      columns={2}
      rows={1}
      spacing={30}
    >
      <styled.div
        key="0"
        width={150}
      >
        <styled.div>
          Some title
        </styled.div>
        <styled.div>
          Some head
        </styled.div>
      </styled.div>
      <styled.div
        key="1"
        width={150}
      >
        <styled.div>
          Other title
        </styled.div>
        <styled.div>
          Other head
        </styled.div>
      </styled.div>
    </Styled(Grid)>
  </styled.div>
</styled.div>
`;
